Update unloadability.md (#15713)
dotnet/docs: docs/standard/assembly/unloadability.md

- Instances of custom non-collectible `AssemblyLoadContext` types created inside of the collectible `AssemblyLoadContext`
- Pending <xref:System.Threading.RegisteredWaitHandle> instances with callbacks set to methods in the custom `AssemblyLoadContext`

Here are hints to finding references stored in stack slots or processor registers that could prevent unloading of your `AssemblyLoadContext`:
- Passing function call results directly to another function may create a reference even though there is no user-created local variable.

> - If a reference to an object was available at any point in a method, the JIT might have decided to keep the it in a stack slot / processor register for as long as it wants in the current function.

Reviewer: ```suggestion
> - If a reference to an object was available at any point in a method, the JIT compiler might keep the reference in a stack slot or processor register for as long as it wants in the current function.
```
